src/utils/glob-resolve.ts

```diff
@@ -0,0 +1,8 @@
+import path from 'node:path'
+
+/**
+ * Wrapper around `path.resolve` that replaces all `path.sep` with `/` to ensure it remains a valid glob pattern.
+ */
+export const globResolve = (...paths: string[]): string => {
+  return path.resolve(...paths).replaceAll(path.sep, '/')
+}
```

Comment:
I am not sure about this. Does that mean we need to implement `globJoin` and `globRelative` in the future as well?

IMHO we might want to use https://www.npmjs.com/package/normalize-path instead of self-baking this one.

Reply:
Hopefully not. I'm not too sure about adding the dep; it's slower and more complex than `replaceAll` for the use case required here, which is almost always just about turning a glob into an absolute path.

If instead we didn't pass in the file location as an absolute path but as a relative path to cwd, we wouldn't need to resolve the glob at all.

Reply:
Though looking at it, assumptions of absolute paths are pretty well baked in.

Reply:
https://github.com/un-ts/eslint-plugin-import-x/blob/1ff90b846d294c54c7965db0ceb481e9bf9ed968/src/rules/no-relative-packages.ts#L16-L18 

oh look!

I agree a dedicated globResolve is a bit pointless, better to just convert to posix paths at the point of minimatching (by moving that function out to a utils folder) _or_ to say that all backslashes are path separators, which is what is currently pushed.